c2corg_ui/static/js/documentediting.js

```diff
@@ -33,7 +33,14 @@ app.documentEditingDirective = function() {
     scope: true,
     controller: 'appDocumentEditingController',
     controllerAs: 'editCtrl',
-    bindToController: true
+    bindToController: true,
+    link: function(scope, el, attrs, ctrl) {
+      // The directive executes after the dom is loaded but we have to wait until the expressions are evaluated
```

Comment:
These 500ms are also to wait for the request which fetches the document data? Can't you set `maxSteps` once you have the data?
https://github.com/c2corg/v6_ui/blob/edit/create-route/waypoint-attributes/c2corg_ui/static/js/documentediting.js#L229

And then do the "calculation" how many steps there are in the controller.